packages/alchemy/src/provider.ts

```diff
@@ -107,6 +131,12 @@ export class AlchemyProvider extends SmartAccountProvider<HttpTransport> {
       );
     }
 
-    return withAlchemyGasManager(this, config);
+    if (this.feeOptsSet) {
```

Comment:
I worry about this a little bit. It'd be nice to just fix the withAlchemyGasManager to not override right? 

Specifically by not calling the code block below conditionally in `withAlchemyGasManager` if the fee ops are set in the config? 
```
  .withFeeDataGetter(async () => ({
     maxFeePerGas: 0n,
     maxPriorityFeePerGas: 0n,
  }))
```

Reply:
curious on @moldy530's thoughts

Reply:
oh wait... I have a thing for this already https://github.com/alchemyplatform/aa-sdk/blob/main/packages/core/src/provider/base.ts#L311

you could potentially just override the sendUserOperation method here to do something like:

```ts
sendUserOperation = async (
    data: UserOperationCallData | BatchUserOperationCallData,
    overrides?: UserOperationOverrides
): Promise<SendUserOperationResult> => {
  super.sendUserOperation(data, overrides ?? this.overrides)
}
```

Reply:
You can't override the fees after the paymaster has signed them

Reply:
This code switches between the two middleware functions based on if the fee opts are set. If no fees are set we should use the middleware where we make gas estimation a no-op.

Not really sure I see another way to do this? The issue isn't really the fee estimation logic, its the removal of the gas estimation logic.

We could also remove the usage of `alchemy_requestGasAndPaymasterAndData` entirely from this code, that would keep this cleaner (but we would lose test coverage there).

Reply:
gotcha yea so requestGasAndPaymaster data overrides all both the fee estimates and gas estimates performed. So if you have any overrides set, then that would mean that those don't get respected if you want to use a paymaster which by default gets gas + fee estimates. 

I'm good with the approach here, it makes sense.